damus/Components/ImageCarousel.swift

```diff
@@ -73,41 +73,73 @@ struct ImageCarousel: View {
-                                .imageContext(.note, disable_animation: disable_animation)
-                                .cancelOnDisappear(true)
-                                .configure { view in
-                                    view.framePreloadCount = 3
-                                }
-                                .imageFill(for: geo.size, max: maxHeight, fill: fillHeight) { fill in
-                                    previews.cache_image_meta(evid: evid, image_fill: fill)
-                                    image_fill = fill
-                                }
-                                .aspectRatio(contentMode: filling ? .fill : .fit)
-                                .tabItem {
-                                    Text(url.absoluteString)
-                                }
-                                .id(url.absoluteString)
+        VStack {
+            TabView(selection: $selectedIndex) {
+                ForEach(urls.indices, id: \.self) { index in
+                    Rectangle()
+                        .foregroundColor(Color.clear)
+                        .overlay {
+                            GeometryReader { geo in
+                                KFAnimatedImage(urls[index])
+                                    .callbackQueue(.dispatch(.global(qos:.background)))
+                                    .backgroundDecode(true)
+                                    .imageContext(.note, disable_animation: disable_animation)
+                                    .cancelOnDisappear(true)
+                                    .configure { view in
+                                        view.framePreloadCount = 3
+                                    }
+                                    .imageFill(for: geo.size, max: maxHeight, fill: fillHeight) { fill in
+                                        previews.cache_image_meta(evid: evid, image_fill: fill)
+                                        image_fill = fill
+                                    }
+                                    .aspectRatio(contentMode: filling ? .fill : .fit)
+                                    .tabItem {
+                                        Text(urls[index].absoluteString)
+                                    }
+                                    .id(urls[index].absoluteString)
+                            }
                         }
-                    }
+                        .tag(index)
+                }
             }
+            .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))
+            .fullScreenCover(isPresented: $open_sheet) {
+                ImageView(urls: urls, disable_animation: disable_animation)
+            }
+            .frame(height: height)
+            .onTapGesture {
+                open_sheet = true
+            }
+            .onChange(of: selectedIndex) { value in
+                            selectedIndex = value
+                        }
```

Comment:
this looks like a no-op to me

Reply:
Not sure why, but it doesn't work if I remove this.